presto-main/src/main/java/com/facebook/presto/tdigest/TDigest.java

```diff
@@ -142,7 +142,7 @@ private TDigest(double compression)
         this.compression = 2 * this.publicCompression;
 
         // having a big buffer is good for speed
-        this.maxBufferSize = 5 * (int) ceil(this.publicCompression + sizeFudge);
+        this.maxBufferSize = 5 * (int) ceil(this.compression + sizeFudge);
```

Comment:
looks like publicCompression is always higher so it can have more bytes, but it should not under allocate.

The integer multiplication can also overflow here and could it cause issues ?

Reply:
`publicCompression` is always 1/2 of `compression`.

The integer multiplication should be fine since it's bounded (can be no larger than 2000).

Reply:
My bad, there is parameter compression, and member variable compression and they are different.